pkg/compactor/compactor.go

```diff
@@ -707,8 +786,9 @@ func (c *Compactor) ownUser(userID string) (bool, error) {
 		return false, nil
 	}
 
-	// Always owned if sharding is disabled.
-	if !c.compactorCfg.ShardingEnabled {
+	// Always owned if sharding is disabled or if using shuffle-sharding as shard ownership
+	// is determined by the shuffle sharding grouper.
+	if !c.compactorCfg.ShardingEnabled || c.compactorCfg.ShardingStrategy == util.ShardingStrategyShuffle {
 		return true, nil
```

Comment:
Every Compactor now "owns" all the users, even if it doesn't actually compact any blocks for most users. One particular example I saw is a huge growth of metadata syncs. Instead, maybe we can compute `ownUser` based on the shuffle shard of a tenant.

Reply:
Quick fix that seems to work: https://github.com/cortexproject/cortex/commit/b0c2c2af681e9a7f1879b88485d8683afbe59bab

Reply:
hey @edma2 thanks for taking a look at the compactor PR. We are actively testing this branch in our beta environment, and I'm glad to hear that it is working for you.

We have a similar fix to your commit, and I can confirm that we also saw an improvement in the metadata syncs. 

However, on the tenant clean up side, we were running into issues where the deleted tenant directory was left dangling. While compactor-A is deleting the deletion markers, compactor-B is also trying to sync the data, and re-uploads the block index. We currently provide an override for the tenant shard size on the clean-up path so that only 1 compactor owns the cleanup for a given tenant.

Reply:
thanks for the tip on the cleanup side. Makes sense that you don't want multiple compactors repeating the same work and potentially conflicting.